graphql_compiler/compiler/ir_lowering_match/utils.py

```diff
@@ -60,6 +85,117 @@ def to_gremlin(self):
         raise NotImplementedError()
 
 
+def filter_edge_field_non_existence(edge_expression):
+    """Return an Expression that is True iff the specified edge (edge_expression) does not exist."""
+    # When an edge does not exist at a given vertex, OrientDB represents that in one of two ways:
+    #   - the edge's field does not exist (is null) on the vertex document, or
+    #   - the edge's field does exist, but is an empty list.
+    # We check both of these possibilities.
+    if not isinstance(edge_expression, (LocalField, SelectEdgeContextField)):
+        raise AssertionError(u'Received invalid edge_expression {} of type {}.'
+                             u'Expected LocalField or SelectEdgeContextField.'
+                             .format(edge_expression, type(edge_expression).__name__))
+    if isinstance(edge_expression, LocalField):
+        if edge_expression.field_name[:3] != u'in_' and edge_expression.field_name[:4] != u'out_':
```

Comment:
can we extract this into a helper function, since it's used in a few different places now?

Reply:
Sure, where else is it being used?

Reply:
nvm, found some